src/Conversion/ONNXToTorch/NN/SoftmaxOp.cpp

```diff
@@ -23,64 +23,53 @@ using namespace mlir::torch;
- * “The operator computes the normalized exponential values for the given
- * input:” “” “ Softmax(input, axis) = Exp(input) / ReduceSum(Exp(input),
- * axis=axis, keepdims=1) “ “” “The input does not need to explicitly be a
- * 2D vector. The "axis" attribute” “indicates the dimension along which 
- * Softmax will be performed.” “The output tensor has the same shape” “and
- * contains the Softmax values of the corresponding input.”
- *
- * Attributes:
- * 	axis	::mlir::IntegerAttr	64-bit signed integer attribute
- *
- *  ONNX Axis attribute Value is map to dimension in torch side.
- *
- * Operands :
- *   input	tensor of 16-bit/32-bit/64-bit float values or
- *   		tensor of bfloat16 type values or memref of any type values
- *   ONNX input is map to input in the torch side.
- *
- * Output   : 
- *   output   	tensor of 16-bit/32-bit/64-bit float values or              
- *              tensor of bfloat16 type values or memref of any type values
- *
- */
-
+/// ONNX Softmax operation 
+/// 
+/// “The operator computes the normalized exponential values for the given
+/// input:” “” “ Softmax(input, axis) = Exp(input) / ReduceSum(Exp(input),
+/// axis=axis, keepdims=1) “ “” “The input does not need to explicitly be a
+/// 2D vector. The "axis" attribute” “indicates the dimension along which 
+/// Softmax will be performed.” “The output tensor has the same shape” “and
+/// contains the Softmax values of the corresponding input.”
+/// 
+/// Attributes:
+/// 	axis	::mlir::IntegerAttr	64-bit signed integer attribute
+/// 
+///  ONNX Axis attribute Value is map to dimension in torch side.
+/// 
+/// Operands :
+///   input	tensor of 16-bit/32-bit/64-bit float values or
+///   		tensor of bfloat16 type values or memref of any type values
+///   ONNX input is map to input in the torch side.
+/// 
+/// Output   : 
+///   output   	tensor of 16-bit/32-bit/64-bit float values or              
+///              tensor of bfloat16 type values or memref of any type values
+ 
 
-class ONNXSoftmaxOpToTorchLowering : public ConversionPattern {
+class ONNXSoftmaxOpToTorchLowering : public OpConversionPattern<ONNXSoftmaxOp> {
 public:
-  ONNXSoftmaxOpToTorchLowering(TypeConverter &typeConverter, 
-	MLIRContext *ctx)
-      : ConversionPattern( typeConverter,
-	mlir::ONNXSoftmaxOp::getOperationName(), 1, ctx) {}
-
-  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
-      ConversionPatternRewriter &rewriter) const final {
-
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(ONNXSoftmaxOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
     Location loc = op->getLoc();
-    mlir::MLIRContext *context =  op->getContext();
-    ONNXSoftmaxOp op1 = llvm::dyn_cast<ONNXSoftmaxOp>(op);
-    ONNXSoftmaxOpAdaptor adapter(op1);
+    MLIRContext *context =  op->getContext();
+    Value input = op.input();
+
+    Torch::ValueTensorType inputType = toTorchType (context, input.getType());
```

Comment:
You can use the `adaptor` directly. Probably `adaptor.X()` to get the corresponding `Torch::ValueTensor`.